packages/react-native/React/CoreModules/RCTDevLoadingView.mm

```diff
@@ -123,36 +123,40 @@ - (void)showMessage:(NSString *)message color:(RCTUIColor *)color backgroundColo
 
     self->_showDate = [NSDate date];
 
-#if !TARGET_OS_OSX // [macOS]
-    UIWindow *mainWindow = RCTKeyWindow();
-    self->_window = [[UIWindow alloc] initWithWindowScene:mainWindow.windowScene];
-    self->_window.windowLevel = UIWindowLevelStatusBar + 1;
-    self->_window.rootViewController = [UIViewController new];
-#else // [macOS
-    self->_window = [[NSWindow alloc] initWithContentRect:NSMakeRect(0, 0, 375, 20)
-                                               styleMask:NSWindowStyleMaskBorderless | NSWindowStyleMaskFullSizeContentView
-                                                 backing:NSBackingStoreBuffered
-                                                   defer:YES];
-    [self->_window setIdentifier:sRCTDevLoadingViewWindowIdentifier];
-#endif // macOS]
-
-    self->_container = [[RCTUIView alloc] init]; // [macOS]
-    self->_container.backgroundColor = backgroundColor;
-    self->_container.translatesAutoresizingMaskIntoConstraints = NO;
-
-    self->_label = [[RCTUILabel alloc] init]; // [macOS]
-    self->_label.translatesAutoresizingMaskIntoConstraints = NO;
-    self->_label.font = [UIFont monospacedDigitSystemFontOfSize:12.0 weight:UIFontWeightRegular];
-    self->_label.textAlignment = NSTextAlignmentCenter;
+    if (!self->label) {
+      self->_label = [[RCTUILabel alloc] init]; // [macOS]
+      self->_label.translatesAutoresizingMaskIntoConstraints = NO;
+      self->_label.font = [UIFont monospacedDigitSystemFontOfSize:12.0 weight:UIFontWeightRegular];
+      self->_label.textAlignment = NSTextAlignmentCenter;
+    }
     self->_label.textColor = color;
     self->_label.text = message;
 
+    if (!self->_container) {
+      self->_container = [[RCTUIView alloc] init]; // [macOS]
+      self->_container.translatesAutoresizingMaskIntoConstraints = NO;
+      [self->_container addSubview:self->_label];
+    }
+    self->_container.backgroundColor = backgroundColor;
+
 #if !TARGET_OS_OSX // [macOS]
-    [self->_window.rootViewController.view addSubview:self->_container];
+    UIWindow *mainWindow = RCTKeyWindow();
+    if (!self->_window) {
+      self->_window = [[UIWindow alloc] initWithWindowScene:mainWindow.windowScene];
+      self->_window.windowLevel = UIWindowLevelStatusBar + 1;
+      self->_window.rootViewController = [UIViewController new];
+      [self->_window.rootViewController.view addSubview:self->_container];
+    }
 #else // [macOS
```

Comment:
While virtuous, I'd prefer the diff only have macOS changes, and keep the original iOS code in tact. I guess we can introduce an [iOS] tag it we really want..

Reply:
Okay, I'll see what I can do!